homeassistant/components/generic_thermostat/climate.py

```diff
@@ -272,6 +273,15 @@ def precision(self):
             return self._temp_precision
         return super().precision
 
+    @property
+    def target_temperature_step(self) -> Optional[float]:
+        """Return the supported step of target temperature."""
+        # Since this integration does not yet have a step size parameter
+        # we have to re-use the precision as the step size for now.
+        if self._temp_precision is not None:
+            return self._temp_precision
+        return super().precision
```

Comment:
Why don't just `return self.precision`?

Reply:
Mainly to stay consistent with other methods such as `min_temp()` and to clearly show that this is coming from the parent and is the default fallback value and not something specifically set by the `GenericThermostat` integration.

Reply:
If you dont intent to call the proper super function (`return super().target_temperature_step`), then this entire function should call `return self.precision` to clearly indicate its using the exact same logic as `precision`

By doing what you are suggesting, you are implicitly hiding any future changes in the super class for when people do not configure `precision` explicitly